### PR TITLE
RFC: eliminate snake case and simplify unambiguous names

### PR DESCRIPTION
**Pull request by shutej**
Hi,

I was reading this package and I noticed a few things:

1. There are instances of snake-case in several places.  This is generally not idiomatic in Go.  If camel case is required for reflection mangling, generally people explicitly call it out in struct tags.
2. A few of the identifiers could be simplified, using the fact that Go generally qualifies imports to disambiguate what they are.  For instance, if there's only one interface in a package, you can just call it `Interface`.

As such, I wanted to send this over for comments.  I realize these would be backward-incompatible changes but I could certainly go on and fix packetbeat and other notable consumers (or perhaps write a `go vet` or some such to clean up examples in folks' private repos).

I also notice that in several places the acronyms are capitalized (`IP`) vs. camelcased (`Ip`), but before fixing those wanted to get some comments.

Jeremy

### CODE REVIEW

**Comment by shutej** [comment]
Oh, one other thing, slices are nil-able types by default, rarely do you need a pointer to a slice.  (Maybe you do in the case I "fixed" but I haven't looked too closely.  Didn't seem like you did, and tests pass.)

File: common/geolite.go [hunk 1]
```diff
@@ -9,27 +9,29 @@ import (
 	"github.com/nranchev/go-libGeoIP"
 )
 
-type Geoip struct {
-	Paths *[]string
```
Comment (tsg) [inline common/geolite.go]: This is the pointer to slice instance you mentioned, right? It was there for separating between "paths" being specified in the configuration file but empty and not being present in the configuration file at all. Not sure if there's a better way to do it.

File: common/bytes.go [hunk 2]
```diff
@@ -8,21 +8,21 @@ import (
 
 // Byte order utilities
 
-func Bytes_Ntohs(b []byte) uint16 {
+func Ntohs(b []byte) uint16 {
```
Comment (tsg) [inline common/bytes.go]: These functions are used from Packetbeat so we'd need to update that as well.

File: outputs/outputs.go [hunk 3]
```diff
@@ -6,34 +6,34 @@ import (
...
-	Filename           string
-	Rotate_every_kb    int
-	Number_of_files    int
-	DataType           string
-	Flush_interval     *int
-	Bulk_size          *int
-	Max_retries        *int
+type Config struct {
+	Enabled           bool
+	SaveTopology      bool
```
Comment (tsg) [inline outputs/outputs.go]: Here we need json keys to keep the old behavior, otherwise we break the compatibility with the configuration file.

Comment (tsg) [inline outputs/outputs.go]: Actually not json keys, but yaml keys probably.

**Comment by tsg** [comment]
Thanks for the pull request. I'm +1 on these changes but we have to be careful with keeping the compatibility for the configuration file and test the changes with Packetbeat.

File: common/geolite.go (hunk 1 above)
Comment (shutej) [inline common/geolite.go]: So there's a distinction between `g.Paths = nil` and `g.Paths = []string{}`.  The former will be true in a configuration file where you `omitempty`.

All this stuff should really be covered by a unit test, I think.  That's the only way I can be relatively sure that this parses the config the way it should.  Maybe there's an example config in another repo you could point to?

File: common/bytes.go (hunk 2 above)
Comment (shutej) [inline common/bytes.go]: Are they used in other places, yet?  If so I could do a `go fix` (messed that up and typed `go vet` before, different beast).

File: outputs/outputs.go (hunk 3 above)
Comment (shutej) [inline outputs/outputs.go]: If you point to an example config our two I'll test this change still works the same.  That's likely to yield the best results.

**Comment by urso** [comment]
I'm a big fan of cleaning this up, but 1. other beats MUST be fixed too (use gorename + PR for all affected beats) and 2. config file parsing must not break (having #127 should be done before).

While getting the naming right, we might want to have a look at go metalinter output too.

**Comment by tsg** [comment]
These changes are happening but step wise in different PRs. As we are planning to merge the beats repositories into a single one this week, I'm closing this one because I don't want it to be implicitly lost.

**Pull request closed without merging**